cosmwasm-plugin/main.go

```diff
@@ -0,0 +1,296 @@
+	templatesFS, err := fs.Sub(templates, "wasm-wiring")
+	if err != nil {
+		fmt.Println("Error:", err)
+		return err
+	}
+
+	// Open the embedded input file
+	inputFile, err := templatesFS.Open(inputFilename)
+
+	if err != nil {
+		return err
+	}
+	defer inputFile.Close() // This line schedules the file to be closed when the function returns
+
+	// Read content from the embedded file
+	sourceContent, err := io.ReadAll(inputFile)
+	if err != nil {
+		return err
+	}
+
+	// Construct the output file path
+	outputPath := filepath.Join(outputDir, outputFilename)
+
+	// Write the embedded content to the output file
+	err = ioutil.WriteFile(outputPath, sourceContent, 0644)
+	if err != nil {
+		return err
+	}
+
+	// Replace chainName in the output file
+	err = replaceWordsInFile(outputPath, "planet", chainName)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Created %s in %s\n", outputFilename, outputDir)
+	return nil
+}
+
+func createNewFiles() error {
+	// Call createFile for each template file
+
+	// Create ante.go based on ante.txt content
+	err := createFile("ante.txt", "app", "ante.go")
+	if err != nil {
+		return err
+	}
+
+	// Create ante2.go based on wasm.txt content
+	err = createFile("wasm.txt", "app", "wasm.go")
+	if err != nil {
+		return err
+	}
+
+	// Create network.go based on network.txt content
+	err = createFile("app.txt", "app", "app.go")
+	if err != nil {
+		return err
+	}
+
+	// Create network.go based on network.txt content
+	err = createFile("simulation_test.txt", "app", "simulation_test.go")
+	if err != nil {
+		return err
+	}
+
+	// Create network.go based on network.txt content
+	err = createFile("network.txt", "testutil/network", "network.go")
+	if err != nil {
+		return err
+	}
+
+	// Create network.go based on network.txt content
+	filePath := filepath.Join("cmd", chainNamed, "cmd")
+	err = createFile("root.txt", filePath, "root.go")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// For future versions..
+func modifyExistingFiles() error {
+	// Modify existingFile1.go
+	content, err := os.ReadFile("existingFile1.go")
+	if err != nil {
+		return err
+	}
+	modifiedContent := string(content) + "\n// Additional content for existingFile1.go"
+	err = os.WriteFile("existingFile1.go", []byte(modifiedContent), 0644)
+	if err != nil {
+		return err
+	}
+
+	// Similarly, modify other existing files...
+
+	return nil
+}
+
+func main() {
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		fmt.Println("Error:", err)
+		return
+	}
+
+	// Extract the chain name from the current directory
+	chainName = filepath.Base(currentDir)
+
+	// Handle potential cases where the chain name contains dashes or underscores
+	chainName = strings.ReplaceAll(chainName, "-", "")
+	chainName = strings.ReplaceAll(chainName, "_", "")
+	chainNamed = chainName + "d"
+
+	//fmt.Println("User's chain name:", chainName)
+
+	pluginMap := map[string]hplugin.Plugin{
+		"cosmwasm-plugin": &plugin.InterfacePlugin{Impl: &p{}},
```

Comment:
Assigning `chainName` to the plugin instance might be a good idea to avoid having the two chain name global variables. A plugin getter could be used for example to get the binary name (`chainNamed`).

Reply:
Implemented changes with this [commit](https://github.com/ignite/plugins/commit/38af11e00e770736f6f3a9bfa37d04dc241bd865)